### PR TITLE
Presentation/labels provider batches

### PR DESCRIPTION
**Pull request by aurislt7**
Updated `LabelsProvider` to send requests in batches

### CODE REVIEW

File: presentation/components/src/presentation-components/labels/LabelsProvider.ts [hunk 1]
```diff
@@ -66,7 +66,18 @@ export class PresentationLabelsProvider implements IPresentationLabelsProvider {
...
+    const labelPromises = new Array<Promise<LabelDefinition[]>>();
+    const tempKeys = [...keys];
+
+    while (tempKeys.length) {
+      const batch = tempKeys.splice(0, DEFAULT_KEYS_BATCH_SIZE);
+      labelPromises.push(Presentation.presentation.getDisplayLabelDefinitions({ imodel: this.imodel, keys: batch }));
+    }
+
+    return (await Promise.all(labelPromises)).reduce((displayValues, labels) => {
+      Array.prototype.push.apply(displayValues, labels.map((def) => def.displayValue));
```
Comment (grigasp) [inline presentation/components/src/presentation-components/labels/LabelsProvider.ts:78]: why not `displayValues.push(...labels.map(...))`?

**Review by grigasp — CHANGES_REQUESTED** [review]
I'd like to see a test that confirms keys are sent in batches

File: presentation/components/src/presentation-components/labels/LabelsProvider.ts (hunk 1 above)
Comment (aurislt7) [reply presentation/components/src/presentation-components/labels/LabelsProvider.ts:78]: Looks better, changed it

**Review by grigasp — APPROVED** [review]

**Review by DanEastBentley — APPROVED** [review]

**Pull request merged**